# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1365442464 14400
#      Mon Apr 08 13:34:24 2013 -0400
# Node ID 7369b210dbd0090ee09a05b400a9c317cbeef677
# Parent  c89699ca005abc08a7884b00bb8d9ee95c92241a
use pointer for terminal inside terminal_dock_widget

* terminal-dock-widget.h, terminal-dock-widget.cc (terminal):
Use pointer.  Change all uses.

diff --git a/libgui/src/terminal-dock-widget.cc b/libgui/src/terminal-dock-widget.cc
--- a/libgui/src/terminal-dock-widget.cc
+++ b/libgui/src/terminal-dock-widget.cc
@@ -23,48 +23,48 @@ along with Octave; see the file COPYING.
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include "terminal-dock-widget.h"
 
 terminal_dock_widget::terminal_dock_widget (QWidget *p)
-  : octave_dock_widget (p), terminal (p)
+  : octave_dock_widget (p), terminal (new QTerminal (p))
 {
-  terminal.setObjectName ("OctaveTerminal");
-  terminal.setFocusPolicy (Qt::StrongFocus);
+  terminal->setObjectName ("OctaveTerminal");
+  terminal->setFocusPolicy (Qt::StrongFocus);
 
   setObjectName ("TerminalDockWidget");
   setWindowIcon (QIcon(":/actions/icons/logo.png"));
   setWindowTitle (tr ("Command Window"));
-  setWidget (&terminal);
+  setWidget (terminal);
 
   connect (parent (), SIGNAL (settings_changed (const QSettings *)),
            this, SLOT (notice_settings (const QSettings *)));
 
   connect (this, SIGNAL (visibilityChanged (bool)),
            this, SLOT (handle_visibility (bool)));
 
   connect (parent (), SIGNAL (relay_command_signal (const QString&)),
            this, SLOT (relay_command (const QString&)));
 
   // Forward signals to QTerminal widget.
 
   connect (this, SIGNAL (notice_settings_signal (const QSettings *)),
-           &terminal, SLOT (notice_settings (const QSettings *)));
+           terminal, SLOT (notice_settings (const QSettings *)));
 
   connect (this, SIGNAL (relay_command_signal (const QString&)),
-           &terminal, SLOT (relay_command (const QString&)));
+           terminal, SLOT (relay_command (const QString&)));
 
   connect (this, SIGNAL (copyClipboard_signal (void)),
-           &terminal, SLOT (copyClipboard (void)));
+           terminal, SLOT (copyClipboard (void)));
 
   connect (this, SIGNAL (pasteClipboard_signal (void)),
-           &terminal, SLOT (pasteClipboard (void)));
+           terminal, SLOT (pasteClipboard (void)));
 }
 
 void
 terminal_dock_widget::notice_settings (const QSettings *settings)
 {
   emit notice_settings_signal (settings);
 }
 
diff --git a/libgui/src/terminal-dock-widget.h b/libgui/src/terminal-dock-widget.h
--- a/libgui/src/terminal-dock-widget.h
+++ b/libgui/src/terminal-dock-widget.h
@@ -58,12 +58,12 @@ signals:
   void relay_command_signal (const QString& command);
 
   void copyClipboard_signal (void);
 
   void pasteClipboard_signal (void);
 
 private:
 
-  QTerminal terminal;
+  QTerminal *terminal;
 };
 
 #endif // TERMINALDOCKWIDGET_H
